File: Python/Problems/Problems/problem3516.py
from .problem import Problem


class Problem3516(Problem):
    def test(self) -> bool:
        x = 2
        y = 7
        z = 4

        expected = 1

        result = self.findClosest(x, y, z)

        return result == expected

    def findClosest(self, x: int, y: int, z: int) -> int:
        dist1 = abs(x - z)
        dist2 = abs(y - z)
        if dist1 == dist2:
            return 0
        return 1 if dist1 < dist2 else 2

File: C++/Problems/Problems/problem_3516.cpp
#include "problem_3516.h"

bool problem_3516::test() {
    auto x = 2;
    auto y = 7;
    auto z = 4;

    auto expected = 1;

    auto result = findClosest(x, y, z);

    return result == expected;
}

int problem_3516::findClosest(int x, int y, int z) {
    auto dist1 = std::abs(x - z);
    auto dist2 = std::abs(y - z);
    if (dist1 == dist2)
        return 0;
    return dist1 < dist2 ? 1 : 2;
}
